### PR TITLE
Properly deal with unicode filenames for source uploads (bug 1193056)

### PR DESCRIPTION
**Pull request by magopian**
Fixes [bug 1193056](https://bugzilla.mozilla.org/show_bug.cgi?id=1193056)

### CODE REVIEW

**Comment by magopian** [comment]
@kmaglione can you please give this one a quick look? I've checked with Jason, we should be able to properly store files on the filesystem with non-ascii characters.

File: apps/versions/views.py [hunk 1]
```diff
@@ -142,6 +142,7 @@ def download_source(request, version_id):
         if not owner_or_unlisted_reviewer(request, version.addon):
             raise http.Http404  # Not listed, not owner or admin.
     res = HttpResponseSendFile(request, version.source.path)
-    name = os.path.basename(version.source.path.replace('"', ''))
-    res['Content-Disposition'] = 'attachment; filename="{0}"'.format(name)
+    name = os.path.basename(version.source.path.replace(u'"', u''))
```
Comment (kmaglione) [inline apps/versions/views.py]: I think you're going to need a `.decode('utf-8')` here. Or an `.encode('utf-8')` and replace stick with 8-bit strings here and below.

**Comment by kmaglione** [comment]
You have failing tests, but otherwise nice work. r+wc

File: apps/versions/tests.py [hunk 2]
```diff
@@ -909,8 +924,14 @@ def test_owner_should_be_allowed(self):
         eq_(response.status_code, 200)
         assert response[settings.XSENDFILE_HEADER]
         assert 'Content-Disposition' in response
-        assert self.filename in response['Content-Disposition']
-        eq_(response[settings.XSENDFILE_HEADER], self.version.source.path)
+        filename = self.filename
+        if not isinstance(filename, unicode):
```
Comment (magopian) [inline apps/versions/tests.py]: Not happy with all those, that I needed to add for the tests to run correctly on travis (they were running fine locally, but raised unicode errors on travis). Not sure why though.

Comment (kmaglione) [inline apps/versions/tests.py]: Why do you need the isinstance check? `.decode` works fine on byte strings.

Comment (kmaglione) [inline apps/versions/tests.py]: OK... It's a bit messy, but I can't think of a way to do it that isn't.

**Comment by kmaglione** [comment]
r+
